homeassistant/components/matter/entity.py

```diff
@@ -110,15 +112,38 @@ async def async_added_to_hass(self) -> None:
 
     async def async_will_remove_from_hass(self) -> None:
         """Run when entity will be removed from hass."""
+        if self._extra_poll_timer:
+            self._extra_poll_timer.cancel()
+            self._extra_poll_timer = None
         for unsub in self._unsubscribes:
             with suppress(ValueError):
                 # suppress ValueError to prevent race conditions
                 unsub()
 
+    async def async_update(self) -> None:
+        """Call when the entity needs to be updated."""
+        if not self.should_poll:
+            return
+        # manually poll/refresh the primary value
+        await self.matter_client.refresh_attribute(
+            self._endpoint.node.node_id,
+            self.get_matter_attribute_path(self._entity_info.primary_attribute),
+        )
+        self._update_from_device()
+
     @callback
     def _on_matter_event(self, event: EventType, data: Any = None) -> None:
-        """Call on update."""
+        """Call on update from the device."""
         self._attr_available = self._endpoint.node.available
+        if self._attr_should_poll:
+            # secondary attribute updated of a polled primary value
+            # enforce poll of the primary value a few seconds later
+            if self._extra_poll_timer:
+                self._extra_poll_timer.cancel()
+            self._extra_poll_timer = self.hass.loop.call_later(
```

Comment:
Use `helpers.event.async_call_later`.

Make a constant for the time to make it easy to patch in tests.

Reply:
The async_call_later only seems to take a coroutine function and not a regular function ?
I want to pass it "async_schedule_update_ha_state" including argument "force_refresh".

I could pass it "async_update_ha_state" instead without any side effects ?

Reply:
Make a callback helper method on the entity that calls `async_schedule_update_ha_state`. The helper should accept one parameter for the time of the event and return `None`.

Reply:
Thanks, first time I had to use this async_call_later helper.
Anyways, I've adjusted te logic a bit to use the helper and define the delay in a constant.